rclpy/src/rclpy/_rclpy.c

```diff
@@ -2224,13 +2229,25 @@ rclpy_take_response(PyObject * Py_UNUSED(self), PyObject * args)
     destroy_ros_message(taken_response);
     if (!pytaken_response) {
       // the function has set the Python error
+      Py_DECREF(pytuple);
       return NULL;
     }
 
-    return pytaken_response;
-  }
-  // if take_response failed, just do nothing
-  Py_RETURN_NONE;
+    PyObject * pysequence = PyLong_FromLongLong(sequence);
+    if (!pysequence) {
+      Py_DECREF(pytaken_response);
+      Py_DECREF(pytuple);
+      return NULL;
+    }
+    PyTuple_SET_ITEM(pytuple, 0, pysequence);
+    PyTuple_SET_ITEM(pytuple, 1, pytaken_response);
+    return pytuple;
+  }
+  Py_INCREF(Py_None);
```

Comment:
I was surprised that Py_None needed to be incref'ed but apparently that's necessary :+1: